### PR TITLE
Add new user syncer

### PR DESCRIPTION
**Pull request by mgloystein**

### CODE REVIEW

File: openrtb_ext/bidders.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ const (
 	BidderSomoaudience   BidderName = "somoaudience"
 	BidderSovrn          BidderName = "sovrn"
 	BidderSonobi         BidderName = "sonobi"
+	BidderSpotx          BidderName = "spotxchange"
```
Comment (jbaldwin) [inline openrtb_ext/bidders.go]: spotx instead of spotxchange?  I'd be consistent and use spotx everywhere

File: endpoints/cookie_sync_test.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ import (
 	"github.com/prebid/prebid-server/adapters/audienceNetwork"
 	"github.com/prebid/prebid-server/adapters/lifestreet"
 	"github.com/prebid/prebid-server/adapters/pubmatic"
+	"github.com/prebid/prebid-server/adapters/spotxchange"
```
Comment (jbaldwin) [inline endpoints/cookie_sync_test.go]: spotx instead of spotxchange?

File: adapters/spotxchange/usersync_test.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"testing"
+	"text/template"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewSpotxSyncer(t *testing.T) {
+	temp := template.Must(template.New("sync-template").Parse("//sync.search.spotxchange.com/partner?adv_id=BUYER_ID&uid=BUYER_USER_ID&img=IMG&redir=REDIR_URL"))
```
Comment (jbaldwin) [inline adapters/spotxchange/usersync_test.go]: how does this prebid server fill out the tokens? the redir param on here is also a bit weird unless if we are trying to dual sync, perhaps we should talk in person?

**Review by jbaldwin — CHANGES_REQUESTED** [review]
I think we need a review or meeting to go over how this works as I don't understand how this can work by only looking at just the changes here

File: adapters/spotxchange/usersync_test.go (hunk 3 above)
Comment (mgloystein) [reply adapters/spotxchange/usersync_test.go]: Probably. I could use some help with a few other pieces to make sure I'm getting things right.

**Review by jbaldwin — APPROVED** [review]

**Pull request merged**